iothub/device/samples/PnpDeviceSamples/TemperatureController/SystemTextJsonSerializer.cs

```diff
@@ -31,6 +31,20 @@ public override T ConvertFromObject<T>(object objectToConvert)
         {
             return DeserializeToType<T>(((JsonElement)objectToConvert).ToString());
         }
+        
+        public override bool TryGetNestedObjectValue<T>(object objectToConvert, string propertyName, out T outValue)
+        {
+            outValue = default;
+            if (objectToConvert == null || string.IsNullOrEmpty(propertyName))
+            {
+                return false;
+            }
+            if (((JsonElement)objectToConvert).TryGetProperty(propertyName, out var element)) {
+                outValue = DeserializeToType<T>(element.GetRawText());
+                return true;
+            }
+            return false;
+        }
```

Comment:
Some questions around adding this method. 

* Does this add too much complexity? 
* Would you decide that this isn't worth the trouble?
* What would you expect to do instead.

Reply:
Add too much complexity for whom? For our impl? For customers who want to implement their own serializer?

I think the resulting user experience is better, and the cost to impl is worth it.

Reply:
The complexity would be for customers who want to implement their own serializer (since this is a part of the PayloadSerializer abstract class). We would provide newtonsoft and system.text samples, so that should help them out; but it is an additional step that they have to account for.